furnace-service-provider/src/main/java/org/jboss/windup/web/furnaceserviceprovider/WebProperties.java

```diff
@@ -16,20 +10,33 @@
 /**
  * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
  */
-@Singleton
-@Startup
 public class WebProperties
 {
     public static final String FURNACE_PROPERTIES = "/windupweb.properties";
     public static final String ADDON_REPOSITORY = "addon.repository";
     public static final String RULES_REPOSITORY = "rules.repository";
 
+    private static WebProperties instance;
     private Properties properties = new Properties();
     private Path addonRepository;
     private Path rulesRepository;
 
-    @Inject
-    private ServletContext servletContext;
+    public static WebProperties getInstance()
+    {
+        if (instance == null)
+        {
+            synchronized (WebProperties.class)
+            {
+                if (instance == null)
```

Comment:
Double-checked locking to reduce synchronization. Although instance should probably be marked volatile as well.